clojure/src/info/lotharschulz/armstrong_numbers/armstrong_numbers.clj

```diff
@@ -0,0 +1,38 @@
+(ns info.lotharschulz.armstrong_numbers.armstrong_numbers
+  (:import [java.Math]))
+
+(defn get-number-of-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns the :getNumberOfDigits of n if n is an integer and greater than -1"
+  (if (zero? n)
+    1
+    (if (pos? n )
+      (+ (int (Math/log10 n )) 1)
+      nil)))
+
+(defn get-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "return number of digits of given int n"
+    (if (zero? n)
+      [0]
+      (if (pos? n)
+        (into [] (map #(Character/getNumericValue %) (str n)))
```

Comment:
> if you specifically want a vector out of (map ...), you can also use (mapv ...) instead of (into [] (map ...))


I was unsure about the best approach reading resources like http://stackoverflow.com/questions/12044181/into-or-vec-converting-sequence-back-to-vector-in-clojure.